mbus/nats_handler.go

```diff
@@ -118,14 +118,27 @@ func (h *natsHandler) Run(handlerFunc boshhandler.Func) error {
 
 	return nil
 }
+func asyncErrorHandler(natsErrorHandlingChan chan string, logger boshlog.Logger, p boshplatform.Platform, connInfo ConnectionInfo) {
+	logger.Debug(natsHandlerLogTag, "Received async nats log")
+	err := p.DeleteARPEntryWithIP(connInfo.IP)
+	if err != nil {
+		logger.Debug(natsHandlerLogTag, fmt.Sprintf("Delete ArpEntries for %v, Error: %v", connInfo.IP, err.Error()))
+	}
```

Comment:
Shouldn't the above be in the loop below?

We think the way `asyncErrorHandler` is currently structured, the ARP flush will occur when the goroutine is first started, rather than anytime an error is received in the `natsErrorHandlingChan`.